### DISCUSSION TITLE
Anyone have a 2D MonoGame.SpriteBatch-like class for Slik.NET?

### DISCUSSION THREAD

**Discussion by Thraka** [dotnet/Silk.NET#548] [Q&A] [answered]
I have a game engine (SadConsole) that was originally built for MonoGame. With the last release I made it generic enough that I can plug in different backends and I created a backend for SFML. The problem, though, is that the unofficial SMFL NuGet bindings and runtime packages are having problems on Linux and require my users additional effort to get things running.

### What I'm looking for

Does anyone have some code that replicates the sprite batching techniques of MonoGame?

### SadConsole

My library is named SadConsole and is hosted at <https://github.com/Thraka/SadConsole>. It's an ANSI/ASCII text UI styled game library that can be used to produce roguelike, ascii, or text games. SadConsole is sprite heavy and uses sprite batching techniques to render 10,000's of sprites.

At its core, it's just a tile engine. You can use different sprite sheets to simulate text/ascii and mix in graphical tiles.

**Comment by ThomasMiz** [comment, marked as answer]
Hello!

I've made a Silk-based library called TrippyGL that provides a class called TextureBatcher. The library works over OpenGL.

Text rendering is also implemented through the TextureBatcher class!

You can find the repo here: https://github.com/ThomasMiz/TrippyGL

Any questions are welcomed :)

**Reply by Thraka** [reply]
Awesome, thank you! I'll try playing with it to see what I can get working :)
